### PR TITLE
[15.0][MIG] sale_coupon_multiplier_free_product: Migration to version 15.0

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
cc@Tecnativa TT37310

@chienandalu @pedrobaeza please review!

### CODE REVIEW

**Review by chienandalu — CHANGES_REQUESTED** [review]
Tests are failing...

File: sale_coupon_multiplier_free_product/models/sale_coupon.py [hunk 1]
```diff
@@ -4,10 +4,13 @@
...
 class SaleCoupon(models.Model):
-    _inherit = "sale.coupon"
+    _inherit = "coupon.coupon"
 
-    def _check_coupon_code(self, order):
-        message = super()._check_coupon_code(order)
+    def _check_coupon_code(self, order_date, partner_id, **kwargs):
+        order = kwargs.get("order")
+        message = super()._check_coupon_code(
+            order.date_order.date(), order.partner_id.id, order=order
```
Comment (chienandalu) [inline sale_coupon_multiplier_free_product/models/sale_coupon.py]: This isn't right. You should pass the args as they come not assuming the values in your override

File: sale_coupon_multiplier_free_product/models/sale_coupon_program.py [hunk 2]
```diff
@@ -75,9 +75,12 @@ def _filter_not_ordered_reward_programs(self, order):
         return programs
 
     def _is_valid_product(self, product):
+        """This method was removed, it's functionallity
```
Comment (chienandalu) [inline sale_coupon_multiplier_free_product/models/sale_coupon_program.py]: Then is useless to override it

File: sale_coupon_multiplier_free_product/models/sale_order.py [hunk 3]
```diff
@@ -141,9 +142,10 @@ def _remove_invalid_reward_lines(self):
                 and x.coupon_program_id in multiple_of_programs_to_remove
             ).unlink()
         # We'll catch the context in the subsequent unlink() method
-        super(
+        res = super(
             SaleOrder, self.with_context(valid_multiple_of_lines=valid_lines)
         )._remove_invalid_reward_lines()
+        return res
```
Comment (chienandalu) [inline sale_coupon_multiplier_free_product/models/sale_order.py]: You don't need to make an assignment if you're going to return that value immediately after it
```suggestion
        return super(
            SaleOrder, self.with_context(valid_multiple_of_lines=valid_lines)
        )._remove_invalid_reward_lines()
```

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: sale_coupon_multiplier_free_product/models/sale_order.py (hunk 3 above)
Comment (stefan-tecnativa) [reply sale_coupon_multiplier_free_product/models/sale_order.py]: All changes done, thank you for all the suggestions.

**Comment by sergio-teruel** [comment]
ping @chienandalu

**Review by chienandalu — APPROVED** [review]
:+1:

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**